provider/marathon/marathon_test.go

```diff
@@ -9,107 +9,78 @@ import (
 )
 
 type fakeClient struct {
 	mocks.Marathon
 }
 
-func newFakeClient(applicationsError bool, applications *marathon.Applications, tasksError bool, tasks *marathon.Tasks) *fakeClient {
+func newFakeClient(applicationsError bool, applications marathon.Applications) *fakeClient {
 	// create an instance of our test object
 	fakeClient := new(fakeClient)
 	if applicationsError {
-		fakeClient.On("Applications", mock.Anything).Return(nil, errors.New("error"))
+		fakeClient.On("Applications", mock.Anything).Return(nil, errors.New("fake Marathon server error"))
 	} else {
-		fakeClient.On("Applications", mock.Anything).Return(applications, nil)
-	}
-	if !applicationsError {
-		if tasksError {
-			fakeClient.On("AllTasks", mock.Anything).Return(nil, errors.New("error"))
-		} else {
-			fakeClient.On("AllTasks", mock.Anything).Return(tasks, nil)
-		}
+		fakeClient.On("Applications", mock.Anything).Return(&applications, nil)
 	}
 	return fakeClient
 }
 
-func TestMarathonLoadConfig(t *testing.T) {
+func TestMarathonLoadConfigAPIErrors(t *testing.T) {
+	fakeClient := newFakeClient(true, marathon.Applications{})
+	provider := &Provider{
+		marathonClient: fakeClient,
+	}
+	actualConfig := provider.loadMarathonConfig()
+	fakeClient.AssertExpectations(t)
+	if actualConfig != nil {
+		t.Errorf("configuration should have been nil, got %v", actualConfig)
+	}
+}
+
+func TestMarathonLoadConfigNonAPIErrors(t *testing.T) {
 	cases := []struct {
-		applicationsError bool
-		applications      *marathon.Applications
-		tasksError        bool
-		tasks             *marathon.Tasks
-		expectedNil       bool
+		desc              string
+		application       marathon.Application
+		task              marathon.Task
 		expectedFrontends map[string]*types.Frontend
 		expectedBackends  map[string]*types.Backend
 	}{
 		{
-			applications:      &marathon.Applications{},
-			tasks:             &marathon.Tasks{},
-			expectedFrontends: map[string]*types.Frontend{},
-			expectedBackends:  map[string]*types.Backend{},
-		},
-		{
-			applicationsError: true,
-			applications:      &marathon.Applications{},
-			tasks:             &marathon.Tasks{},
-			expectedNil:       true,
-			expectedFrontends: map[string]*types.Frontend{},
-			expectedBackends:  map[string]*types.Backend{},
-		},
-		{
-			applications:      &marathon.Applications{},
-			tasksError:        true,
-			tasks:             &marathon.Tasks{},
-			expectedNil:       true,
-			expectedFrontends: map[string]*types.Frontend{},
-			expectedBackends:  map[string]*types.Backend{},
-		},
-		{
-			applications: &marathon.Applications{
-				Apps: []marathon.Application{
-					{
-						ID:     "/test",
-						Ports:  []int{80},
-						Labels: &map[string]string{},
-					},
-				},
+			desc: "simple application",
+			application: marathon.Application{
+				Ports:  []int{80},
+				Labels: &map[string]string{},
 			},
-			tasks: &marathon.Tasks{
-				Tasks: []marathon.Task{
+			task: marathon.Task{
```

Comment:
Could you reduce test definition with some helpers builder?

```golang
// ...
    task: aMarathonTask(withDefaultIPAddress),
// ...
```
```golang

func aMarathonTask(builders ...func(marathon.Task)) marathon.Task {
	task := marathon.Task{
		Host:  "localhost",
		Ports: []int{80},
	}

	for _, builder := range builders {
		builder(task)
	}

	return task
}

func withDefaultIPAddress(task marathon.Task) {
	task.IPAddresses =  []*marathon.IPAddress{
		{
			IPAddress: "127.0.0.1",
			Protocol:  "tcp",
		},
	}
}
```
I think you do the same for `marathon.Application`, `types.Backend`, `types.Frontend`.